### PR TITLE
Merge branch 'merge_cosmicfish'

### PR DESCRIPTION
**Pull request by Sefa76**
Changes proposed in this pull request:
Merged remote cosmicfish repo into cosmicfishpie
        Mainly changes to the plotting routine
        Added new external file for nonlinear power spectra
        New options for spectroscopic error. Removed infamous h-bug

### CODE REVIEW

**Comment by Sefa76** [comment]
This has now the newest additions to cosmicfish.
- Fishconsumer module 
- additional option to add your own colors to the plotting routines

This also fixes old problems
- Removed any deprecated functions with slur-adjacent names
- Toned down the printing when plotting fishers

File: cosmicfishpie/LSSsurvey/spectro_obs.py [hunk 1]
```diff
@@ -801,8 +812,12 @@ def observed_Pgg(self, z, k, mu, b_i=None):
...
         tstart = time()
 
-        k = self.k_units_change(k)  # has to be done before spec_err and AP
-        error_z = self.spec_err_z(z, k, mu)  # before rescaling of k,mu by AP
+        if self.kh_rescaling_beforespecerr:
+            k = self.k_units_change(k)  # has to be done before spec_err and AP
+            error_z = self.spec_err_z(z, k, mu)  ##before rescaling of k,mu by AP
+        else:
+            error_z = self.spec_err_z(z, k, mu)  ##before rescaling of k,mu by AP
+            k = self.k_units_change(k)  # has to be done before spec_err and AP
```
Comment (santiagocasas) [inline cosmicfishpie/LSSsurvey/spectro_obs.py:820]: Leave also a comment here....

**Review by santiagocasas — CHANGES_REQUESTED** [review]
Address the little changes of the bug and that should be all fine! thanks!

File: cosmicfishpie/LSSsurvey/spectro_obs.py [hunk 2]
```diff
@@ -265,6 +265,10 @@ def activate_terms(self):
     def set_spectro_specs(self):
         """Updates the spectroscopic redshift error"""
         self.dz_err = cfg.specs["spec_sigma_dz"]
+        self.dz_type = cfg.specs["spec_sigma_dz_type"]
+        self.kh_rescaling = cfg.specs["spec_khrescale"]
+        self.kh_rescaling_beforespecerr = cfg.specs["spec_khrescale_beforespecerr"]
```
[the remark this answers is not in the document]
Comment (Sefa76) [reply cosmicfishpie/LSSsurvey/spectro_obs.py:270]: After looking at it a bit longer i dont thin its a good idea to have the bug in the survey specifications
I would rather have it in options

File: cosmicfishpie/LSSsurvey/spectro_obs.py [hunk 3]
```diff
@@ -344,8 +348,8 @@ def kper(self, z, k, mu):
 
     def k_units_change(self, k):
         """
-        Function that rescales the k-array, when the kmax-kmin integration units are fixed in h/Mpc,
-        while the rest of the code is defined in 1/Mpc.
+        Function that rescales the k-array, when asked for.
+        The code is defined everzwhere in 1/Mpc so a rescaling would be wwrong.
```
Comment (santiagocasas) [inline cosmicfishpie/LSSsurvey/spectro_obs.py:352]: a typo here in z/y

File: cosmicfishpie/LSSsurvey/spectro_obs.py [hunk 4]
```diff
@@ -266,10 +267,13 @@ def set_spectro_specs(self):
...
         self.dz_err = cfg.specs["spec_sigma_dz"]
         self.dz_type = cfg.specs["spec_sigma_dz_type"]
-        self.kh_rescaling = cfg.specs["spec_khrescale"]
-        self.kh_rescaling_beforespecerr = cfg.specs["spec_khrescale_beforespecerr"]
         ## constant, z-dependent
 
+        # These bugs are intentionally left in, in order to reproduce old results.
+        # The reallity is that they are not to be here.
+        self.kh_rescaling_bug = cfg.settings["kh_rescaling_bug"]
+        self.kh_rescaling_beforespecerr_bug = cfg.settings["kh_rescaling_beforespecerr_bug"]
```
Comment (santiagocasas) [inline cosmicfishpie/LSSsurvey/spectro_obs.py:275]: ok, agree with having this in specs, you are right!

File: cosmicfishpie/LSSsurvey/spectro_obs.py [hunk 5]
```diff
@@ -357,7 +361,12 @@ def k_units_change(self, k):
         float, numpy.ndarray
             wavenumbers in un units of h ref/Mpc
         """
-        if self.kh_rescaling:
+        if self.kh_rescaling_bug:
+            warnings.warn(
+                "You requested to do an additional unpysical rescaling of the wavenumbers (h-bug).",
+                category=RuntimeWarning,
```
Comment (santiagocasas) [inline cosmicfishpie/LSSsurvey/spectro_obs.py:367]: nice!

File: cosmicfishpie/LSSsurvey/spectro_obs.py [hunk 6]
```diff
@@ -812,12 +821,17 @@ def observed_Pgg(self, z, k, mu, b_i=None):
             print("    Computing Pgg for {}".format(self.observables))
         tstart = time()
 
-        if self.kh_rescaling_beforespecerr:
-            k = self.k_units_change(k)  # has to be done before spec_err and AP
-            error_z = self.spec_err_z(z, k, mu)  ##before rescaling of k,mu by AP
+        if self.kh_rescaling_beforespecerr_bug:
+            # In this case the the h-bug is only applied before computing the resolution supressuion
```
Comment (santiagocasas) [inline cosmicfishpie/LSSsurvey/spectro_obs.py:825]: typo: suppression

**Review by santiagocasas — CHANGES_REQUESTED** [review]
thanks for the changes, after fixing the typos, this should be all good!

**Review by santiagocasas — APPROVED** [review]
Good!

**Comment by santiagocasas** [comment]
well done!

**Pull request merged**